crates/tx5-go-pion/src/evt.rs

```diff
@@ -56,150 +58,282 @@ pub enum PeerConnectionEvent {
-        // TODO!!! MEORY LEAK
-        // we need else cases througout here, if there isn't a
-        // registered callback, we need to call _free functions
-        // on incoming events like DataChannel and OnMessage buffers.
-        API.on_event(|sys_evt| match sys_evt {
-            SysEvent::Error(_error) => (),
-            SysEvent::PeerConICECandidate {
-                peer_con_id,
-                candidate,
-            } => {
-                let maybe_cb =
-                    MANAGER.lock().peer_con.get(&peer_con_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(PeerConnectionEvent::ICECandidate(GoBuf(candidate)));
-                }
-            }
-            SysEvent::PeerConStateChange {
-                peer_con_id,
-                peer_con_state,
-            } => {
-                let maybe_cb =
-                    MANAGER.lock().peer_con.get(&peer_con_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(PeerConnectionEvent::State(
-                        PeerConnectionState::from_raw(peer_con_state),
-                    ));
-                }
-            }
-            SysEvent::PeerConDataChan {
-                peer_con_id,
-                data_chan_id,
-            } => {
-                let maybe_cb =
-                    MANAGER.lock().peer_con.get(&peer_con_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(PeerConnectionEvent::DataChannel(DataChannelSeed::new(
-                        data_chan_id,
-                    )));
-                }
-            }
-            SysEvent::DataChanClose(data_chan_id) => {
-                let maybe_cb =
-                    MANAGER.lock().data_chan.get(&data_chan_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(DataChannelEvent::Close);
-                }
-            }
-            SysEvent::DataChanOpen(data_chan_id) => {
-                let maybe_cb =
-                    MANAGER.lock().data_chan.get(&data_chan_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(DataChannelEvent::Open);
-                }
-            }
-            SysEvent::DataChanMessage {
-                data_chan_id,
-                buffer_id,
-            } => {
-                let maybe_cb =
-                    MANAGER.lock().data_chan.get(&data_chan_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(DataChannelEvent::Message(GoBuf(buffer_id)));
-                }
-            }
-            SysEvent::DataChanBufferedAmountLow(data_chan_id) => {
-                let maybe_cb =
-                    MANAGER.lock().data_chan.get(&data_chan_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(DataChannelEvent::BufferedAmountLow);
-                }
-            }
+        API.on_event(move |sys_evt| {
+            let _ = send.blocking_send(sys_evt);
```

Comment:
We're using this channel to translate from the synchronous golang FFI events into asynchronous rust. While this represents a bottleneck, the `manager_access!` macro should start killing off connections if the app is too slow to process incoming events within 18ms.